lightning/src/offers/invoice_request.rs

```diff
@@ -500,7 +524,11 @@ impl UnsignedInvoiceRequest {
 
 		invoice_request_tlv_stream.write(&mut bytes).unwrap();
 
-		const EXPERIMENTAL_TLV_ALLOCATION_SIZE: usize = 0;
+		// Allocate sufficient capacity for experimental TLV fields to avoid reallocations.
+		// The new fields (invreq_contact_secret: ~48 bytes, invreq_payer_offer: ~116 bytes,
+		// invreq_payer_bip_353_name: ~116 bytes) total ~280 bytes, with 600 providing headroom
```

Comment:
No need to be this conservative, I don't think. Things work if we underestimate, its just extra allocations. We can be a bit tighter (eg 300b).